.github/workflows/continuous-integration.yml

```diff
@@ -33,8 +34,20 @@ jobs:
             setup.py
             requirements-dev.txt
 
-      - name: Execute linters and test suites
-        run: ./scripts/cibuild
+      - name: Install package
+        run: pip install .
+
+      - name: Install dev requirements
+        run: pip install -r requirements-dev.txt
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files
+
+      - name: Run pytest
+        run: pytest -Werror -s --block-network --cov pystac_client --cov-report term-missing
```

Comment:
What is the `-Werror` option? Not finding that in the docs.

Reply:
Converts all warnings into errors (e.g. to catch places where we hit deprecations): https://docs.python.org/3/using/cmdline.html#cmdoption-W

Reply:
Thanks.